src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,59 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.dto.ProductResponseDto;
+import gift.service.ProductService;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @PostMapping
+    public ResponseEntity<ProductResponseDto> createProduct(
+        @RequestBody ProductRequestDto requestDto
+    ) {
+        return new ResponseEntity<>(productService.createProduct(requestDto), HttpStatus.OK);
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ProductResponseDto>> findAllProducts() {
+        return new ResponseEntity<>(productService.findAllProducts(), HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ProductResponseDto> findProductById(@PathVariable Long id) {
+        return new ResponseEntity<>(productService.findProductById(id), HttpStatus.OK);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<ProductResponseDto> updateProduct(
+        @PathVariable Long id,
+        @RequestBody ProductRequestDto requestDto
+    ) {
+        return new ResponseEntity<>(productService.updateProduct(id, requestDto), HttpStatus.OK);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteProduct(@PathVariable Long id) {
+        productService.deleteProduct(id);
+
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
```

Comment:
```suggestion
        return ResponseEntity.noContent().build();
```
마찬가지로 이렇게 작성할 수 있어요!